src/lib/mdns/minimal/Server.cpp

```diff
@@ -156,7 +161,7 @@ CHIP_ERROR ServerBase::Listen(chip::Inet::InetLayer * inetLayer, ListenIterator
 
     while (it->Next(&interfaceId, &addressType))
     {
-        ReturnErrorCodeIf(endpointIndex >= mEndpointCount, CHIP_ERROR_NO_MEMORY);
+        ReturnErrorCodeIf(endpointIndex >= mEndpointCount, autoShutdown.ReturnSuccess());
```

Comment:
We ran out of indices, if we happen to need those network endpoints this would silently return 'everything ok'.

Need at least a log if we decided to not error-out.

Reply:
Would also seem more readable if we 'break' here instead of fast-returning 'success'.

Reply:
_Very_ good catch.  This change was not supposed to be here; it was left over from my initial debugging.  Reverted this part, and re-verified that we still succeed in not running out of endpoints on Mac, so don't hit this case anyway for the moment.